frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.test.tsx

```diff
@@ -73,55 +167,121 @@ describe('AccessControlTab', () => {
+        partyTypesAllowed: { ...mockAppMetadata.partyTypesAllowed },
+      }),
+    );
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('settings_modal.loading_content')),
+    );
+    const checkboxes = screen.queryAllByRole('checkbox');
+    expect(checkboxes).toHaveLength(5);
+    checkboxes.forEach((c) => expect(c).toBeTruthy());
+  });
+
+  it('should render all checkboxes labels', async () => {
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('settings_modal.loading_content')),
+    );
+
+    const bankruptcyEstateCheckbox = screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_bankruptcy_estate'),
     });
-    expect(bankruptcyEstateCheckbox).toBeChecked();
+    expect(bankruptcyEstateCheckbox).toBeInTheDocument();
 
-    const organisationCheckbox = screen.getByRole('checkbox', {
+    const organisationCheckbox = screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_organisation'),
     });
-    expect(organisationCheckbox).not.toBeChecked();
+    expect(organisationCheckbox).toBeInTheDocument();
 
-    const personCheckbox = screen.getByRole('checkbox', {
+    const personCheckbox = screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_person'),
     });
-    expect(personCheckbox).not.toBeChecked();
+    expect(personCheckbox).toBeInTheDocument();
 
-    const subUnitCheckbox = screen.getByRole('checkbox', {
+    const subUnitCheckbox = screen.getByRole('row', {
       name: textMock('settings_modal.access_control_tab_option_sub_unit'),
     });
-    expect(subUnitCheckbox).not.toBeChecked();
+    expect(subUnitCheckbox).toBeInTheDocument();
   });
 
-  it('handles checkbox changes', async () => {
+  it('renders the documentation link with the correct text', async () => {
+    render();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('settings_modal.loading_content')),
+    );
+
+    const documentationLink = screen.getByText(
+      textMock('settings_modal.access_control_tab_option_access_control_docs_link_text'),
+    );
+    expect(documentationLink).toBeInTheDocument();
+  });
+
+  it('renders the modal when user tries to uncheck the last checked checkbox, and close it when clicking on close button', async () => {
     const user = userEvent.setup();
     await resolveAndWaitForSpinnerToDisappear();
-
-    const organisationCheckboxBefore = screen.getByRole('checkbox', {
-      name: textMock('settings_modal.access_control_tab_option_organisation'),
+    render();
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    await waitFor(async () => {
+      const checkboxes = screen.getAllByRole('checkbox');
+      expect(checkboxes).toHaveLength(5);
+      checkboxes.forEach((checkbox, index) => {
+        if (index < 4) {
+          user.click(checkbox);
+        }
+      });
```

Comment:
Is this necessary? We can pass mocked data where only one is checked (which I think is the case for the already mocked queryresult) and simply uncheck the checked box by its name instead of indexes.

Reply:
Agree, i will change it after i fix the bug